paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -86,20 +88,17 @@ class PaparazziPlugin : Plugin<Project> {
 
       supportedPlugins.forEach { plugin ->
         project.plugins.withId(plugin) {
-          val variants = when (val extension = project.extensions.getByType(TestedExtension::class.java)) {
-            is LibraryExtension -> extension.libraryVariants
-            is BaseAppModuleExtension -> extension.applicationVariants
-            is DynamicFeatureExtension -> extension.applicationVariants
-            // exhaustive to avoid potential breaking changes in future AGP releases
-            else -> throw IllegalStateException("${extension.javaClass.name} from $plugin is not supported in Paparazzi")
+          val extension = project.extensions.getByType(AndroidComponentsExtension::class.java)
+          check(extension is LibraryAndroidComponentsExtension || extension is ApplicationAndroidComponentsExtension || extension is DynamicFeatureAndroidComponentsExtension) {
```

Comment:
Is this still necessary?

Is there maybe a one-to-one mapping between plugin ID and `androidComponents { }` type we can rely on?

Reply:
Sorry, I didn't include enough context in the highlight:
```kotlin
val supportedPlugins = listOf("com.android.application", "com.android.library", "com.android.dynamic-feature")
...
supportedPlugins.forEach { plugin ->
    project.plugins.withId(plugin) {
        val extension = project.extensions.getByType(AndroidComponentsExtension::class.java)
        // At this point we can be quite sure extension is not `TestAndroidComponentsExtension`,
        // because that would imply someone applied `com.android.test` plugin,
        // which is mutually exclusive to all the others in `supportedPlugins`.
```